### ISSUE TITLE
"build" folder created in the 3-rd party libraries on deploy

### ISSUE THREAD

**Issue by vfilimonov** [aws/chalice#1415]
Hello

on deploy lambda functions via chalice, a new `build` folder is created in each library, which is installed from "requirements.txt". I've noticed it since several libs in "requirements" I install from a local folder rather than from pip.

I don't know if it is an expected behavior or not. Apparently it was introduced in one of recent versions and happened after upgrade to 0.13.1/0.14.

chalice version: 0.14

**Comment by jamesls** [comment]
There was a packaging bug fix as part of the 1.14 release that could be related.  Could you give me more info or a specific package that has this behavior.  I tried it with a requirements file of:

```
/Users/name/Source/chalice
```

And after running `chalice package` I don't see a build directory:

```
$ ls -la ~/Source/chalice | grep build
```

**Comment by vfilimonov** [comment]
Hm, I could not reproduce it either. When it happened I put this `build` folder in `.gitignore` of all my local repos so I don't even have visibility on when and how it was gone.

Let me then close the issue for now, and if it happen again, I'll reopen it with more details.


P.S. Just not to forget, I'm putting here a minimal example that should have reproduced it (but as of now there's no extra `build` folder in `local_pkg`):
```
#!/bin/bash

mkdir local_pkg
echo "from setuptools import setup, find_packages; setup(name='local_pkg', packages=find_packages())" > local_pkg/setup.py
mkdir local_pkg/local_pkg
touch local_pkg/local_pkg/__init__.py
pip install -e local_pkg

chalice new-project temp
cd temp
echo "../local_pkg" >> requirements.txt
chalice package out

pip uninstall local-pkg
```
